src/components/AttachmentPicker/index.native.js

```diff
@@ -318,65 +242,118 @@ class AttachmentPicker extends Component {
      * @param {Object} item - an item from this.menuItemData
      * @param {Function} item.pickAttachment
      */
-    selectItem(item) {
+    function selectItem(item) {
         /* setTimeout delays execution to the frame after the modal closes
          * without this on iOS closing the modal closes the gallery/camera as well */
-        this.onModalHide = () =>
+        onModalHide.current = () =>
             setTimeout(
                 () =>
                     item
                         .pickAttachment()
-                        .then(this.pickAttachment)
+                        .then(pickAttachment)
                         .catch(console.error)
-                        .finally(() => delete this.onModalHide),
+                        .finally(() => delete onModalHide.current),
                 200,
             );
 
-        this.close();
+        close();
+    }
+
+    function attachKeyboardListener() {
+        const shortcutConfig = CONST.KEYBOARD_SHORTCUTS.ENTER;
+        keyboardListener.current = KeyboardShortcut.subscribe(
+            shortcutConfig.shortcutKey,
+            () => {
+                if (focusedIndex === -1) {
+                    return;
+                }
+                selectItem(menuItemData[focusedIndex]);
+                setFocusedIndex(-1); // Reset the focusedIndex on selecting any menu
+            },
+            shortcutConfig.descriptionKey,
+            shortcutConfig.modifiers,
+            true,
+        );
     }
 
+    function removeKeyboardListener() {
+        if (!keyboardListener.current) {
+            return;
+        }
+        keyboardListener.current();
+    }
+
+    useEffect(() => {
+        // When selecting an image on a native device, it would be redundant to have a second option for choosing a document,
+        // so it is excluded in this case.
+        if (props.type === CONST.ATTACHMENT_PICKER_TYPE.IMAGE) {
+            return;
+        }
+
+        setMenuItemData((oldMenuItemData) => [
+            ...oldMenuItemData,
+            {
+                icon: Expensicons.Paperclip,
+                textTranslationKey: 'attachmentPicker.chooseDocument',
+                pickAttachment: () => showDocumentPicker(),
+            },
+        ]);
+
+        return () => {
+            removeKeyboardListener();
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    useEffect(() => {
+        if (isVisible) {
+            attachKeyboardListener();
+        } else {
+            removeKeyboardListener();
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [isVisible]);
```

Comment:
You can wrapper these functions with `useCallback`, to have a correct deps. array.